src/components/modals/loginModal.js

```diff
@@ -11,14 +11,26 @@ export const LoginModal = ({ users }) => {
     setShowSignUpModal,
   } = useContext(ModalsContext);
 
+  const modalRef = useRef();
+
+  useEffect(() => {
+    document.addEventListener("mousedown", (e) => {
+      if (!modalRef.current.contains(e.target)) {
```

Comment:
Check, but I think that you don’t need to check if current ref is same as target